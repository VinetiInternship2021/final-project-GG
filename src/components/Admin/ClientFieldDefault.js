import React from 'react';

const ClientFieldDefault = () => (
  <div key={0} className="list-group-item list-group-item-action ">
    <div className="d-flex w-100 justify-content-between">
      <h5 className="mb-1">
        No people
      </h5>
    </div>
  </div>
);

export default ClientFieldDefault;
