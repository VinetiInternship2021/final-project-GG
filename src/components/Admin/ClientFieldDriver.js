/* eslint-disable */
import React from 'react';
import PropTypes from 'prop-types';
import VerifyButton from './verification/VerifyButton';

const ClientFieldDriver = ({
  driver: {
    id, phoneNumber, firstName, lastName, isActive, isVerifiedByAdmin,
  },
  onClick,
  filter,
}) => {
  const verifiedClassName = isVerifiedByAdmin ? 'bg-success' : 'bg-danger';
  const activeClassName = isActive && isVerifiedByAdmin ? 'bg-success' : 'bg-danger';

  return (
    <div key={id} className="list-group-item list-group-item-action">
      <div className="d-flex w-100 justify-content-between">
        <h5 className="mb-1">
          { `${firstName} ${lastName}` }
        </h5>
        <span className={`badge ${verifiedClassName}`}>
          {isVerifiedByAdmin ? 'Confirmed' : 'Unconfirmed'}
        </span>
      </div>
      <p className="mb-1">
        { `Phone number ${phoneNumber}`}
      </p>
      <small className={`badge ${activeClassName}`}>
        {isActive ? 'Active' : 'Inactive'}
      </small>
      {(!isVerifiedByAdmin && !filter) ? (
        <VerifyButton
          onClick={onClick}
          text="Verify"
        />
      ) : ''}
    </div>
  );
};

ClientFieldDriver.propTypes = {
  driver: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default ClientFieldDriver;
