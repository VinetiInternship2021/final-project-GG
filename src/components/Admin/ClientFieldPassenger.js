import React from 'react';
import PropTypes from 'prop-types';

const ClientFieldPassenger = ({
  passenger: {
    id, phoneNumber, firstName, lastName,
  },
}) => (
  <div key={id} className="list-group-item list-group-item-action ">
    <div className="d-flex w-100 justify-content-between">
      <h5 className="mb-1">
        { `${firstName} ${lastName}` }
      </h5>
      <small> </small>
    </div>
    <p className="mb-1">
      { `Phone number ${phoneNumber}`}
    </p>
  </div>
);

ClientFieldPassenger.propTypes = {
  passenger: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default ClientFieldPassenger;
