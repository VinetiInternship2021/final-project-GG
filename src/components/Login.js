import React from 'react';
import { useHistory } from 'react-router-dom';
import { buttons, appRoutes } from '../utils/configs';

const Login = () => {
    let history = useHistory();
    const onLogin = (button) => {
        let path = ''
        switch (button) {
            case 'Client':
                path = appRoutes.loginClient
                break;
            case 'Driver':
                path = appRoutes.loginDriver
                break;
            default:
                path = ''
        }
        history.push(path)
    }

    const button = buttons.map((button) => {
        return (
            <div key={button} >
                <button onClick={() => onLogin(button)
                }
                        className="btn btn-outline-success mb-1 w-50"
                        type="submit">{button}</button>
                <br />
            </div >
        )
    })

    return (
        <div className="text-center border position-absolute top-50 start-50 translate-middle" style={{ width: "200px", height: "180px" }}>
            <h5 className="mt-2">Login</h5>
            <p>Select user type</p>
            {button}
        </div>
    )
}

export default Login