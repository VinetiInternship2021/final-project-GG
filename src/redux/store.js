import { createStore } from 'redux';
import {
  ACTION_CHANGE_ALERT, ACTION_CHANGE_LOADING,
  ACTION_CHANGE_LOGGED_IN, CREATE_CAR_TYPE,
} from './actionTypes';

const initState = {
  isLoading: true,
  loggedIn: false,
  userType: '',
  userId: '',
  alert: '',
  carType: '',
};

const rootReducer = (state = initState, action) => {
  switch (action.type) {
    case ACTION_CHANGE_LOGGED_IN:
      return {
        ...state,
        isLoading: action.payload.isLoading,
        loggedIn: action.payload.loggedIn,
        userType: action.payload.userType,
        userId: action.payload.userId,
        alert: '',
      };
    case ACTION_CHANGE_LOADING:
      return {
        ...state,
        isLoading: action.payload.isLoading,
      };
    case ACTION_CHANGE_ALERT:
      return {
        ...state,
        alert: action.payload.alert,
      };
    case CREATE_CAR_TYPE:
      return {
        ...state,
        carType: action.payload.type,
      };
    default:
      return state;
  }
};
export default createStore(rootReducer);
