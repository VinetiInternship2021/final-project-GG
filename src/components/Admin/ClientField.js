import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { mapStateToProps } from '../../redux/actions';
import ClientFieldDriver from './ClientFieldDriver';
import ClientFieldPassenger from './ClientFieldPassenger';
import ClientFieldDefault from './ClientFieldDefault';

const ClientField = ({ usersList, client, setState }) => {
  let fields;
  switch (client) {
    case 'passengers':
      try {
        fields = usersList.users.map(({ passenger }) => (
          <ClientFieldPassenger key={passenger.id} passenger={passenger} />
        ));
      } catch (err) {
        console.log(err);
      }
      break;
    case 'drivers':
      try {
        fields = usersList.users.map(({ driver }) => (
          <ClientFieldDriver
            setState={setState}
            key={driver.id}
            driver={driver}
          />
        ));
      } catch (err) {
        console.log(err);
      }
      break;
    default:
      fields = [false].map(() => <ClientFieldDefault />);
  }

  return (
    <div className="list-group">
      {fields}
    </div>
  );
};

ClientField.propTypes = {
  usersList: PropTypes.objectOf(PropTypes.any).isRequired,
  client: PropTypes.string.isRequired,
  setState: PropTypes.func.isRequired,
};

export default connect(mapStateToProps)(ClientField);
