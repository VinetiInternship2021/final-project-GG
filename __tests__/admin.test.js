import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import {
  configure, shallow, render, mount,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import Login from '../src/components/Login';
import ClientFieldDriver from '../src/components/Admin/ClientFieldDriver';
import ClientFieldPassenger from '../src/components/Admin/ClientFieldPassenger';
import ClientFieldDefault from '../src/components/Admin/ClientFieldDefault';
import store from '../src/redux/store';
import ClientField from '../src/components/Admin/ClientField';

describe('Admin testing sets (loggedOut)', () => {
  configure({ adapter: new Adapter() });
  const wrappedComponent = ({ children }) => (
    <Provider store={store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  );

  it('Render correctly Login', () => {
    const component = render(<Login />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Render ClientFieldDriver correctly', () => {
    const driver = {
      id: 1,
      phoneNumber: '094808489',
      firstName: 'Artyom',
      lastName: 'Kosakyan',
      isActive: false,
      isVerifiedByAdmin: false,
    };
    const component = render(<ClientFieldDriver driver={driver} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Render ClientFieldPassenger correctly', () => {
    const passenger = {
      id: 1,
      phoneNumber: '094808489',
      firstName: 'Artyom',
      lastName: 'Kosakyan',
    };
    const component = render(<ClientFieldPassenger passenger={passenger} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Render ClientFieldDefault correctly', () => {
    const component = render(<ClientFieldDefault />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Render ClientField correctly', () => {
    const match = {
      params: {
        client: 'Driver',
      },
    };
    const usersList = {
      users: [{
        id: '1',
        phoneNumber: '094808489',
        firstName: 'Artyom',
        lastName: 'Kosakyan',
        isActive: false,
        isVerifiedByAdmin: false,
        carLevel: 'econom',
      },
      {
        id: '2',
        phoneNumber: '094808489',
        firstName: 'Artyom',
        lastName: 'Kosakyan',
        isActive: false,
        isVerifiedByAdmin: false,
        carLevel: 'econom',
      }],
    };
    const component = <ClientField client={match.params.client} usersList={usersList} />;
    const WrapComponent = shallow(wrappedComponent(component));
    expect(shallowToJson(WrapComponent)).toMatchSnapshot();
  });

  it('Render ClientField Drivers list correctly', () => {
    const client = 'drivers';
    const usersList = {
      users: [{
        driver: {
          id: '1',
          phoneNumber: '094808489',
          firstName: 'Artyom',
          lastName: 'Kosakyan',
          isActive: false,
          isVerifiedByAdmin: false,
          carLevel: 'econom',
        },
      },
      {
        driver: {
          id: '2',
          phoneNumber: '077808489',
          firstName: 'Artyom',
          lastName: 'Kosakyan',
          isActive: false,
          isVerifiedByAdmin: false,
          carLevel: 'business',
        },
      }],
    };

    const component = <ClientField client={client} usersList={usersList} />;
    const mountedComponent = mount(component, { wrappingComponent: wrappedComponent });
    const fields = mountedComponent.find('list-fields');
    expect(fields).toHaveLength(2);
  });
});
