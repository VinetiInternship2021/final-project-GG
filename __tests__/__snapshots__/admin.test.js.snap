// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin testing sets (loggedOut) Render ClientField correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <BrowserRouter>
    <Connect(ClientField)
      client="Driver"
      usersList={
        Object {
          "users": Array [
            Object {
              "carLevel": "econom",
              "firstName": "Artyom",
              "id": "1",
              "isActive": false,
              "isVerifiedByAdmin": false,
              "lastName": "Kosakyan",
              "phoneNumber": "094808489",
            },
            Object {
              "carLevel": "econom",
              "firstName": "Artyom",
              "id": "2",
              "isActive": false,
              "isVerifiedByAdmin": false,
              "lastName": "Kosakyan",
              "phoneNumber": "094808489",
            },
          ],
        }
      }
    />
  </BrowserRouter>
</ContextProvider>
`;

exports[`Admin testing sets (loggedOut) Render ClientFieldDefault correctly 1`] = `null`;

exports[`Admin testing sets (loggedOut) Render ClientFieldDriver correctly 1`] = `null`;

exports[`Admin testing sets (loggedOut) Render ClientFieldPassenger correctly 1`] = `null`;

exports[`Admin testing sets (loggedOut) Render correctly Login 1`] = `null`;
